test/sql/json/read_json_dates.test

```diff
@@ -0,0 +1,89 @@
+# name: test/sql/json/read_json_dates.test
+# description: Read json files - date detection
+# group: [json]
+
+require json
+
+# create date and timestamp tables
+statement ok
+create table date_test as select '1996/03/27'::DATE d
+
+statement ok
+create table timestamp_test as select '1996-03-27 07:42:33'::TIMESTAMP t
+
+# test all supported date formats
+foreach date_format '%m-%d-%Y' '%m-%d-%y' '%d-%m-%Y' '%d-%m-%y' '%Y-%m-%d' '%y-%m-%d'
+
+statement ok
+copy (select d from date_test) to '__TEST_DIR__/my_file.json' (format json, dateformat ${date_format})
+
+# auto-detect
+query II
+select typeof(d), d from '__TEST_DIR__/my_file.json'
+----
+DATE	1996-03-27
+
+# forced format read_json
+query II
+select typeof(d), d from read_json('__TEST_DIR__/my_file.json', columns={d: 'DATE'}, dateformat=${date_format})
+----
+DATE	1996-03-27
+
+# forced format COPY
+statement ok
+drop table if exists date_copy_test
+
+statement ok
+create table date_copy_test (d date)
+
+statement ok
+copy date_copy_test from '__TEST_DIR__/my_file.json' (format json, dateformat ${date_format})
+
+query II
+select typeof(d), d from date_copy_test
+----
+DATE	1996-03-27
+
+endloop
+
+# test all supported timestamp formats (hacky way to do foreach parameters that need spaces in them)
+foreach a,b,c '%Y-%m-%d,%H:%M:%S.%f,' '%m-%d-%Y,%I:%M:%S,%p' '%m-%d-%y,%I:%M:%S,%p' '%d-%m-%Y,%H:%M:%S,' '%d-%m-%y,%H:%M:%S,' '%Y-%m-%d,%H:%M:%S,' '%y-%m-%d,%H:%M:%S,'
+
+statement ok
+copy (select t from timestamp_test) to '__TEST_DIR__/my_file.json' (format json, timestampformat ${a} ${b} ${c})
+
+# auto-detect
+query II
+select typeof(t), t from '__TEST_DIR__/my_file.json'
+----
+TIMESTAMP	1996-03-27 07:42:33
+
+# forced format read_json
+query II
+select typeof(t), t from read_json('__TEST_DIR__/my_file.json', columns={t: 'TIMESTAMP'}, timestamp_format=${a} ${b} ${c})
+----
+TIMESTAMP	1996-03-27 07:42:33
+
+# forced format COPY
+statement ok
+drop table if exists timestamp_copy_test
+
+statement ok
+create table timestamp_copy_test (t timestamp)
+
+statement ok
+copy timestamp_copy_test from '__TEST_DIR__/my_file.json' (format json, timestampformat ${a} ${b} ${c})
```

Comment:
For CSV and Parquet files the format is automatically inferred from the filename (e.g. if you do `COPY x TO 'y.parquet'` it will automatically infer `parquet`) - does this work with JSON as well?

Reply:
It does not - good catch. I've added this to `transform_copy.cpp` :)